### ISSUE TITLE
Cookie storage example fails with error `sessionString is undefined`

### ISSUE THREAD

**Issue by jacopo-j** [jpramosi/geckordp#12]
Hello, I'm trying to run the `cookie_storage.py` example script, but it seems to fail out of the box. Here are the console logs after enabling debug mode:

```
2023-10-15 23:57:41,394 [geckordp][DEBUG] - wait_process_loaded(): expired=False
2023-10-15 23:57:41,407 [geckordp][DEBUG] - connect():
2023-10-15 23:57:41,407 [geckordp][DEBUG] - __connect(): Queue read task
2023-10-15 23:57:41,407 [geckordp][DEBUG] - __connect(): Run IO loop
2023-10-15 23:57:41,407 [geckordp][DEBUG] - __open_connection(): Try to open connection
2023-10-15 23:57:41,412 [geckordp][DEBUG] - __open_connection(): Start listening
2023-10-15 23:57:41,422 [geckordp][DEBUG] - __read(): message complete (226 == 226)
2023-10-15 23:57:41,422 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "from": "root",
  "applicationType": "browser",
  "testConnectionPrefix": "server1.conn0.",
  "traits": {
    "networkMonitor": true,
    "resources": {
      "extensions-backgroundscript-status": true
    },
    "workerConsoleApiMessagesDispatchedToMainThread": true
  }
}
2023-10-15 23:57:41,422 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
2023-10-15 23:57:41,422 [geckordp][DEBUG] - __sync_send_receive():
2023-10-15 23:57:41,422 [geckordp][INFO] - __send(): REQUEST->
{
  "to": "root",
  "type": "listTabs"
}
2023-10-15 23:57:41,425 [geckordp][DEBUG] - __read(): message complete (292 == 292)
2023-10-15 23:57:41,425 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "tabs": [
    {
      "actor": "server1.conn0.tabDescriptor1",
      "browserId": 3,
      "browsingContextID": 4,
      "isZombieTab": false,
      "outerWindowID": 8,
      "selected": true,
      "title": "SameSite Cookies Tester",
      "traits": {
        "watcher": true,
        "supportsReloadDescriptor": true
      },
      "url": "https://samesitetest.com/cookies/set"
    }
  ],
  "from": "root"
}
2023-10-15 23:57:41,425 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
2023-10-15 23:57:41,425 [geckordp][DEBUG] - __sync_send_receive():
2023-10-15 23:57:41,425 [geckordp][INFO] - __send(): REQUEST->
{
  "to": "server1.conn0.tabDescriptor1",
  "type": "getWatcher",
  "isServerTargetSwitchingEnabled": null,
  "isPopupDebuggingEnabled": null
}
2023-10-15 23:57:41,428 [geckordp][DEBUG] - __read(): message complete (596 == 596)
2023-10-15 23:57:41,429 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "actor": "server1.conn0.watcher2",
  "traits": {
    "frame": true,
    "process": true,
    "worker": true,
    "resources": {
      "console-message": true,
      "css-change": true,
      "css-message": true,
      "document-event": true,
      "Cache": true,
      "cookies": true,
      "error-message": true,
      "extension-storage": true,
      "indexed-db": true,
      "local-storage": true,
      "session-storage": true,
      "platform-message": true,
      "network-event": true,
      "network-event-stacktrace": true,
      "reflow": true,
      "stylesheet": true,
      "source": true,
      "thread-state": true,
      "server-sent-event": true,
      "websocket": true,
      "tracing-state": true,
      "last-private-context-exit": true
    }
  },
  "from": "server1.conn0.tabDescriptor1"
}
2023-10-15 23:57:41,429 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
2023-10-15 23:57:41,429 [geckordp][DEBUG] - __sync_send_receive():
2023-10-15 23:57:41,429 [geckordp][INFO] - __send(): REQUEST->
{
  "to": "server1.conn0.watcher2",
  "type": "watchTargets",
  "targetType": "frame"
}
2023-10-15 23:57:41,431 [geckordp][DEBUG] - __read(): message complete (33 == 33)
2023-10-15 23:57:41,431 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "from": "server1.conn0.watcher2"
}
2023-10-15 23:57:41,431 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
2023-10-15 23:57:41,431 [geckordp][DEBUG] - __sync_send_receive():
2023-10-15 23:57:41,431 [geckordp][INFO] - __send(): REQUEST->
{
  "to": "server1.conn0.watcher2",
  "type": "watchResources",
  "resourceTypes": [
    "cookies"
  ]
}
2023-10-15 23:57:41,436 [geckordp][DEBUG] - __read(): message complete (384 == 384)
2023-10-15 23:57:41,436 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "type": "resource-available-form",
  "resources": [
    {
      "actor": "server1.conn0.cookies3",
      "hosts": {
        "https://samesitetest.com": []
      },
      "traits": {
        "supportsAddItem": true,
        "supportsRemoveItem": true,
        "supportsRemoveAll": true,
        "supportsRemoveAllSessionCookies": true
      },
      "resourceType": "cookies",
      "resourceId": "cookies-2147483650",
      "resourceKey": "cookies",
      "browsingContextID": 4
    }
  ],
  "from": "server1.conn0.watcher2"
}
2023-10-15 23:57:41,436 [geckordp][DEBUG] - __handle_events(): [server1.conn0.watcher2][resource-available-form] handled
2023-10-15 23:57:41,436 [geckordp][DEBUG] - __read(): message complete (33 == 33)
2023-10-15 23:57:41,436 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "from": "server1.conn0.watcher2"
}
2023-10-15 23:57:41,437 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
2023-10-15 23:57:41,437 [geckordp][DEBUG] - __sync_send_receive():
2023-10-15 23:57:41,437 [geckordp][INFO] - __send(): REQUEST->
{
  "to": "server1.conn0.cookies3",
  "type": "getStoreObjects",
  "host": "https://samesitetest.com",
  "names": null,
  "options": {}
}
2023-10-15 23:57:41,438 [geckordp][DEBUG] - __read(): message complete (181 == 181)
2023-10-15 23:57:41,438 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "from": "server1.conn0.cookies3",
  "error": "TypeError",
  "message": "sessionString is undefined",
  "fileName": "resource://devtools/shared/natural-sort.js",
  "lineNumber": 50,
  "columnNumber": 5
}
2023-10-15 23:57:41,438 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
2023-10-15 23:57:41,438 [geckordp][ERROR] - __sync_send_receive(): Error on request:
{'to': 'server1.conn0.cookies3', 'type': 'getStoreObjects', 'host': 'https://samesitetest.com', 'names': None, 'options': {}}
{
  "from": "server1.conn0.cookies3",
  "error": "TypeError",
  "message": "sessionString is undefined",
  "fileName": "resource://devtools/shared/natural-sort.js",
  "lineNumber": 50,
  "columnNumber": 5
}
{
  "from": "server1.conn0.cookies3",
  "error": "TypeError",
  "message": "sessionString is undefined",
  "fileName": "resource://devtools/shared/natural-sort.js",
  "lineNumber": 50,
  "columnNumber": 5
}
2023-10-15 23:57:41,438 [geckordp][DEBUG] - __sync_send_receive():
2023-10-15 23:57:41,438 [geckordp][INFO] - __send(): REQUEST->
{
  "to": "server1.conn0.cookies3",
  "type": "editItem",
  "data": {
    "host": "https://samesitetest.com",
    "key": "uniqueKey",
    "field": "value",
    "oldValue": "",
    "newValue": "my_new_value",
    "items": {}
  }
}
2023-10-15 23:57:41,439 [geckordp][DEBUG] - __read(): message complete (33 == 33)
2023-10-15 23:57:41,439 [geckordp][INFO] - __print_response(): RESPONSE<-
{
  "from": "server1.conn0.cookies3"
}
2023-10-15 23:57:41,439 [geckordp][DEBUG] - __handle_single_request(): response valid, set result
Traceback (most recent call last):
  File "cookie_storage.py", line 200, in <module>
    main()
  File "cookie_storage.py", line 152, in main
    _data = stores_update_fut.result(1.0)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/concurrent/futures/_base.py", line 460, in result
    raise TimeoutError()
concurrent.futures._base.TimeoutError
```

I'm using Python 3.10.8 and Firefox 115.3.1esr (64 bit) on macOS 10.14.

Do you have any guidance for debugging the script and making it work? Thanks.

**Comment by jpramosi** [comment]
Thank you for your report, I can confirm that the example is not working.
At the moment you can replace
`cookie_actor.get_store_objects(host)`
with
`cookie_actor.get_store_objects(host, options={"sessionString": ""})`
to workaround this issue.

In the next release I will refresh the actors, diagrams and examples.

**Comment by jacopo-j** [comment]
Thank you! I can confirm that your workaround works.

**Comment by jpramosi** [comment]
Fixed in [ae59c3d4a9db47e9e00ac35dba50a3070758c481](https://github.com/jpramosi/geckordp/commit/ae59c3d4a9db47e9e00ac35dba50a3070758c481).
